code/modules/power/apc.dm

```diff
@@ -783,66 +783,63 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/power/apc/auto_name, APC_PIXEL_OFFSET
 		return
 	if(!ishuman(user))
 		return
+	var/mob/living/carbon/human/apc_interactor = user
+	var/obj/item/organ/stomach/ethereal/maybe_ethereal_stomach = apc_interactor.getorganslot(ORGAN_SLOT_STOMACH)
+	if(!istype(maybe_ethereal_stomach))
+		togglelock(user)
 	else
-		var/mob/living/carbon/human/H = user
-		var/obj/item/organ/stomach/maybe_stomach = H.getorganslot(ORGAN_SLOT_STOMACH)
-		if(!istype(maybe_stomach, /obj/item/organ/stomach/ethereal))
+		var/obj/item/organ/stomach/ethereal/stomach = maybe_ethereal_stomach
```

Comment:
this cast is redundant, you've already got maybe_ethereal_stomach that you can just use directly